geo/src/algorithm/haversine_closest_point.rs

```diff
@@ -0,0 +1,579 @@
+use crate::{haversine_distance::HaversineDistance, Bearing};
+use crate::{Closest, Contains};
+use crate::{CoordsIter, GeoFloat, HaversineDestination, Point, MEAN_EARTH_RADIUS};
+use geo_types::{
+    Coord, Geometry, GeometryCollection, Line, LineString, MultiLineString, MultiPoint,
+    MultiPolygon, Polygon, Rect, Triangle,
+};
+
+use num_traits::FromPrimitive;
+
+/// Closest point on a geometry from a point using great circles.
+///
+/// https://edwilliams.org/avform147.htm#XTE
+///
```

Comment:
Maybe it's worth mentioning that this trait only makes sense for coordinates in a spherical coordinate systems, like WGS84?

Reply:
Done.